core/runtime/src/main/java/io/quarkus/runtime/configuration/HashSetFactory.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.runtime.configuration;
+
+import java.util.HashSet;
+import java.util.function.IntFunction;
+
+/**
+ * A helper class to produce a right-sized hash set.
+ */
+public final class HashSetFactory<T> implements IntFunction<HashSet<T>> {
+    private static final HashSetFactory<?> INSTANCE = new HashSetFactory<>();
+
+    private HashSetFactory() {
+    }
+
+    public HashSet<T> apply(final int value) {
+        return new HashSet<>(value);
```

Comment:
I don't think it works as you think it is: the number of initial elements is dependent of the fill factor. Right now, you will always create a slightly undersized map.

Reply:
Not sure, but we use the ArrayList one in Quarkus Build time Config. 
@dmlloyd care to comment here?

Reply:
Thanks for the explanation!

Reply:
@geoand see this comment in HV where we try to size properly all the collections that are in the metadata: https://github.com/hibernate/hibernate-validator/blob/master/engine/src/main/java/org/hibernate/validator/internal/util/CollectionHelper.java#L130

It's not an issue with `ArrayList` as it sizes the array exactly with the size you pass. In any case, it's a detail.

Reply:
I can add same method if you want, it's not a problem obviously :)

Reply:
Added :)